hikari/utilities/iterators.py

```diff
@@ -201,6 +205,21 @@ class LazyIterator(typing.Generic[ValueT], abc.ABC):
 
     __slots__: typing.Sequence[str] = ()
 
+    def chunk(self, chunk_size: int) -> LazyIterator[typing.Sequence[ValueT]]:
+        """Return results in chunks of up to `chunk_size` amount of entries.
+
+        Parameters
+        ----------
+        chunk_size : int
+            The limit for how many results should be returned in each chunk.
+
+        Returns
+        -------
+        LazyIterator[ValueT]
+            `LazyIterator` that emits each chunked sequence.
```

Comment:
This isn't `ValueT` is it? It is `Sequence[ValueT]`
```suggestion
        LazyIterator[typing.Sequence[ValueT]]
            `LazyIterator` that emits zero or more sequences containing the given 
            `chunk_size` of items.
```